api/config/v1/config.go

```diff
@@ -53,28 +52,21 @@ type Flags struct {
 
 // parseConfig parses a config file as either YAML of JSON and unmarshals it into a Config struct.
 func parseConfig(configFile string) (*Config, error) {
-	reader, err := os.Open(configFile)
+	configYaml, err := os.ReadFile(configFile)
 	if err != nil {
 		return nil, fmt.Errorf("error opening config file: %v", err)
 	}
-	defer reader.Close()
 
-	config, err := parseConfigFrom(reader)
+	config, err := parseConfigFrom(configYaml)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing config file: %v", err)
 	}
 
 	return config, nil
 }
 
-func parseConfigFrom(reader io.Reader) (*Config, error) {
+func parseConfigFrom(configYaml []byte) (*Config, error) {
```

Comment:
Could we keep this using a reader? This will minimize the diff as the change should then be limited to this function. Also, looking at the `ioutil.ReadAll` documentation this mentions that this has been moved to `io.ReadAll` as of Go 1.16. Does this mean that the changes in this commit are redundant (or covered by default Golang version update)?

Reply:
`os.ReadFile` was returning a `[]byte` directly from the filename passed to it so I couldn't keep the reader. but as you said `io.ReadAll` is available in 1.17 so I completely reverted the changes. 

I was building the repo since the beginning with the assumption that the Dockerfile.ubuntu (I found the build steps in the README file, so it might be good to note that it's meant 1.17+ only for the next ones) was correct in using golang 1.15, which is not the case anymore as you said earlier. I simply didn't connect the dots afterward to realize that I should pick function availability from 1.17/1.16 in the docs. 

For my defense, my primary objective with your repo was to implement GPU sharing, and as I found an MR (https://gitlab.com/nvidia/kubernetes/device-plugin/-/merge_requests/86) doing just that I was simply focused in porting it to the latest version, making it work, and testing it on our k8s clusters (which was a success and it's going live in production by Monday :-) ).